### PR TITLE
[INTEL MKL] Enable mkl group conv

### PR DESCRIPTION
**Pull request by Solaryee**
Group conv is already enabled in CPU. This PR enables mkl group conv.

### CODE REVIEW

**Comment by gbaned** [comment]
@penpornk  Can you please review this PR ? Thanks!

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and sorry for the delay! I have a minor nit.

File: tensorflow/core/kernels/mkl/mkl_conv_ops.h [hunk 1]
```diff
@@ -238,6 +240,15 @@ class MklDnnConvUtil {
...
         mkldnn_sizes[MKL_GROUP_FILTER_DIM_W] = filter_cols;
 
+        *filter_dims = mkldnn_sizes;
+      } else if (*is_grouped_convolution) {
+        std::vector<MKLDNN_SIZE_DTYPE> mkldnn_sizes(5, -1);
+        mkldnn_sizes[MKL_GROUP_FILTER_DIM_G] = group_count;
+        mkldnn_sizes[MKL_GROUP_FILTER_DIM_O] = filter_out_depth / group_count;
+        mkldnn_sizes[MKL_GROUP_FILTER_DIM_I] = filter_in_depth;
+        mkldnn_sizes[MKL_GROUP_FILTER_DIM_H] = filter_rows;
+        mkldnn_sizes[MKL_GROUP_FILTER_DIM_W] = filter_cols;
```
[the remark this answers is not in the document]
Comment (ShengYang1) [reply tensorflow/core/kernels/mkl/mkl_conv_ops.h:251]: Thanks for review. Yes, we can directly set `filter_dims` to remove redundant vector copy. Also I have added a TODO and will submit another PR.

**Review by penpornk — APPROVED** [review]
Thank you for the change!

**Comment by penpornk** [comment]
It looks like the [Linux GPU](https://source.cloud.google.com/results/invocations/bc5f88db-c0bb-429f-b5e2-7ac1b70fe5bd/targets) failure is related to this PR. Could you please help take a look at `//tensorflow/python/kernel_tests:conv_ops_test_gpu`? Thank you!

**Comment by ShengYang1** [comment]
> It looks like the [Linux GPU](https://source.cloud.google.com/results/invocations/bc5f88db-c0bb-429f-b5e2-7ac1b70fe5bd/targets) failure is related to this PR. Could you please help take a look at `//tensorflow/python/kernel_tests:conv_ops_test_gpu`? Thank you!

Looks like Eigen conv backward doesn't support group conv, so I removed the changes for gradient test case.

**Review by penpornk — APPROVED** [review]
Thank you for the fix!

**Pull request merged**